### PR TITLE
2827 table create visual regression test fixes

### LINKED ISSUE #2827: [Table] Create visual regression test
Visual regression test are needed on the table component. In order to limit the amount of test we run we will want to create a Table instance that has as many permutations as possible. If two different instances are needed to capture all the functionality let's include them both in one test that snapshots both at once. 

Since there could be differences between headless and headed instance of the test runner you the `onlyOn` method from `@cypress/skip-test` package to skip visual regression test  during headed runs. CardCodeEditor e2e test have example of this.

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #2827

Summary
Adds visual regression testing to the table. I used 3 table instances in separate visual snapshots to cover as much UI as possible. Separate snapshots was needed since I needed to capture UI elements that are only visible on click/hover.

Change List (commits, features, bugs, etc)

- refactored table test helper code so that it could be share between tests
- added tests (See attached excel file for what is covered in each table instance)
[IBM PAL table visual regression testing.xlsx](https://github.com/carbon-design-system/carbon-addons-iot-react/files/7205615/IBM.PAL.table.visual.regression.testing.xlsx)
- FIX where I had to set the environment HEIGHT to the value of the width in order to get the viewport working. This also affected the CardCodeEditor snap which actually got the width defined by the test (minus 1px which seems to be happening in some tests). This is still very confusing, but at least it is working now.

Acceptance Test (how to verify the PR)
Do we capture enough and the right stuff?

Regression Test (how to make sure this PR doesn't break old functionality)
Old unit tests still working as expected

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/components/Table/Table.test.jsx [hunk 1]
```diff
@@ -1369,7 +1298,7 @@ describe('Table', () => {
       />
     );
 
-    expect(screen.queryAllByTestId('table-head--overflow').length).toBe(5);
+    expect(screen.queryAllByTestId('table-head--overflow').length).toBe(8);
```
Comment (davidicus) [inline packages/react/src/components/Table/Table.test.jsx:1301]: Should we have any changes to existing unit test like this?

**Review by davidicus — APPROVED** [review]
Looks good other than the changed test case.

File: packages/react/src/components/Table/Table.test.jsx (hunk 1 above)
Comment (bjornalm) [reply packages/react/src/components/Table/Table.test.jsx:1301]: In general no, but this stems from that I wanted to share the test data between the unit and the e2e tests and for the e2e I needed 3 additional columns so I added those to the `getColumns` function in the Table.test.helpers.jsx. Hence we went from 5 to 8 columns in our tests and I had to adjust this one unit test.

File: packages/react/src/components/Table/Table.test.jsx [hunk 2]
```diff
@@ -173,7 +102,7 @@ describe('Table', () => {
   const expandedData = [
     {
       rowId: 'row-1',
-      content: <RowExpansionContent rowId="row-1" />,
+      content: <div style={{ padding: '8px' }}>Expnded content</div>,
```
Comment (kevinsperrine) [inline packages/react/src/components/Table/Table.test.jsx:105]:
```suggestion
      content: <div style={{ padding: '8px' }}>Expanded content</div>,
```

**Review by kevinsperrine — APPROVED** [review]
Looks good. I wonder why snapshot 1 failed and the others didn't?

File: packages/react/src/components/Table/Table.test.jsx (hunk 2 above)
Comment (bjornalm) [reply packages/react/src/components/Table/Table.test.jsx:105]: Thanks, good catch. I need to work on my spelling.  

Regarding the failed test, I'm looking into it but I think it is related to the "load more" PR that we just merged.

Comment (bjornalm) [reply packages/react/src/components/Table/Table.test.jsx:105]: Ahh.. no of course. It was your fix to the Aggregation column. In the old screenshot it was in the wrong column, that is how I discovered it in the first place.

**Pull request merged**